src/parser/shaman/enhancement/modules/talents/IceStrike.tsx

```diff
@@ -0,0 +1,75 @@
+import React from 'react';
+
+import SPELLS from 'common/SPELLS/index';
+import Analyzer, { SELECTED_PLAYER } from 'parser/core/Analyzer';
+import Statistic from 'interface/statistics/Statistic';
+import STATISTIC_ORDER from 'interface/others/STATISTIC_ORDER';
+import STATISTIC_CATEGORY from 'interface/others/STATISTIC_CATEGORY';
+import Events from 'parser/core/Events';
+import BoringSpellValueText from 'interface/statistics/components/BoringSpellValueText';
+import { formatNumber } from 'common/format';
+import SpellUsable from 'parser/shared/modules/SpellUsable';
+
+/**
+ * Strike your target with an icy blade, dealing (65% of Attack power)
+ * Frost damage and snaring them by 50% for 6 sec.
+ *
+ * Successful Ice Strikes reset the cooldown of your Flame Shock and
+ * Frost Shock spells.
+ *
+ * Example Log:
+ */
+class IceStrike extends Analyzer {
+  static dependencies = {
+    spellUsable: SpellUsable,
+  };
+  protected spellUsable!: SpellUsable;
+
+  protected casts: number = 0;
+  protected cooldownReduced: number = 0;
+
+  constructor(options: any) {
+    super(options);
+
+    if (!this.selectedCombatant.hasTalent(SPELLS.ICE_STRIKE_TALENT.id)) {
+      this.active = false;
+    }
+
+    this.addEventListener(
+      Events.damage.by(SELECTED_PLAYER)
+        .spell(SPELLS.ICE_STRIKE_TALENT),
+      this.onIceStrikeDamage,
+    );
+  }
+
+  onIceStrikeDamage() {
+    this.casts += 1;
+
+    if (this.spellUsable.isOnCooldown(SPELLS.FLAME_SHOCK.id)) {
+      const cd = this.spellUsable.cooldownRemaining(SPELLS.FLAME_SHOCK.id);
```

Comment:
again, not sure why this needs to be put into a constant if the only thing you are doing is adding it to cooldownReduced